src/ert/config/design_matrix.py

```diff
@@ -107,12 +106,12 @@ def merge_with_other(self, dm_other: DesignMatrix) -> None:
             raise ConfigValidationError.from_collected(errors)
 
         try:
-            self.design_matrix_df = pd.concat(
+            self.design_matrix_df = pl.concat(
                 [
                     self.design_matrix_df,
-                    dm_other.design_matrix_df.drop(list(common_keys), axis=1),
+                    dm_other.design_matrix_df.select(pl.exclude(list(common_keys))),
                 ],
-                axis=1,
+                how="horizontal",
```

Comment:
not sure about this one. In another PR I do some concat and use "diagonal" to get the full join. Horizontal is only in case that you add new columns - which is the case here I guess. On the other side, there could be new rows defined in dm_other, but maybe this is checked somewhere else before we do the merge.